### PR TITLE
Regression data ax

### PR DESCRIPTION
**Pull request by furkansahin**
Ready for review

### CODE REVIEW

File: src/test/regress/sql/multi_behavioral_analytics_basics.sql [hunk 1]
```diff
@@ -12,9 +12,8 @@ FROM (
     FROM users_table AS u,
          events_table AS e
     WHERE u.user_id = e.user_id
-      AND u.user_id >= 10
-      AND u.user_id <= 25
-      AND e.event_type IN (100, 101, 102)
+      AND u.user_id = 1
```
Comment (metdos) [inline src/test/regress/sql/multi_behavioral_analytics_basics.sql]: I think we are losing some tests here. For us `>=` and `<=` are not same with` =`. Do you have any reason to convert them?

Comment (furkansahin) [reply src/test/regress/sql/multi_behavioral_analytics_basics.sql]: This was what I worried. Okay, I will find another way around

File: src/test/regress/sql/multi_behavioral_analytics_basics.sql [hunk 2]
```diff
@@ -219,9 +210,8 @@ TRUNCATE agg_results;
 INSERT INTO agg_results(user_id)
 SELECT user_id
 FROM users_table
-WHERE (value_1 = 10
-       OR value_1 = 11
-       OR value_1 = 12)
+WHERE (value_1 >= 1
```
Comment (metdos) [inline src/test/regress/sql/multi_behavioral_analytics_basics.sql]: Here we converted OR to AND. Again, this runs different parts of the code. Why did we change this?

Comment (furkansahin) [reply src/test/regress/sql/multi_behavioral_analytics_basics.sql]: I tried to put a filter to the results. Going to add ORs back

File: src/test/regress/sql/multi_behavioral_analytics_basics.sql [hunk 3]
```diff
@@ -406,14 +396,14 @@ FROM
    users_table
         JOIN 
    (SELECT  
-      ma.user_id, (GREATEST(coalesce(ma.value_4 / 250, 0.0) + GREATEST(1.0))) / 2 AS prob
+      ma.user_id, (GREATEST(coalesce(ma.value_4, 0.0) + GREATEST(1.0))) / 2 AS prob
```
Comment (metdos) [inline src/test/regress/sql/multi_behavioral_analytics_basics.sql]: similarly why we ditched `/ 250`.

I think you get the idea :)

File: src/test/regress/expected/multi_behavioral_analytics_basics.out [hunk 4]
```diff
@@ -12,25 +12,23 @@ FROM (
...
   GROUP BY user_id
 ) q;
 -- get some statistics from the aggregated results to ensure the results are correct
 SELECT count(*), count(DISTINCT user_id), avg(user_id) FROM agg_results;
- count | count |         avg         
--------+-------+---------------------
-     5 |     5 | 15.6000000000000000
+ count | count |          avg           
+-------+-------+------------------------
+     1 |     1 | 1.00000000000000000000
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_basics.out]: if possible it is better to have more than 1 row for aggregates

File: src/test/regress/expected/multi_behavioral_analytics_basics.out [hunk 5]
```diff
@@ -245,14 +238,14 @@ SELECT count(*), count(DISTINCT user_id), avg(user_id) FROM agg_results;
 TRUNCATE agg_results;
 INSERT INTO agg_results(user_id, value_2_agg)
 SELECT user_id, value_2 FROM users_table WHERE
-  value_1 > 101 AND value_1 < 110
-  AND value_2 >= 5
-  AND EXISTS (SELECT user_id FROM events_table WHERE event_type>101  AND event_type < 110 AND value_3 > 100 AND user_id=users_table.user_id);
+  value_1 > 1 AND value_1 < 3
+  AND value_2 >= 3
+  AND EXISTS (SELECT user_id FROM events_table WHERE event_type>1  AND event_type < 3 AND value_3 > 3 AND user_id=users_table.user_id);
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_basics.out]: maybe we can keep range larger to include more rows

File: src/test/regress/expected/multi_partitioning.out [hunk 6]
```diff
@@ -700,38 +700,38 @@ FROM
                    FROM 
                     partitioned_events_table as  "events"
                    WHERE 
-                    event_type IN (10, 11, 12, 13, 14, 15))
+                    event_type = 1 )
```
Comment (metdos) [inline src/test/regress/expected/multi_partitioning.out]: also better to keep using IN

File: src/test/regress/expected/multi_behavioral_analytics_basics.out [hunk 7]
```diff
@@ -225,16 +222,15 @@ TRUNCATE agg_results;
 INSERT INTO agg_results(user_id)
 SELECT user_id
 FROM users_table
-WHERE (value_1 = 10
-       OR value_1 = 11
-       OR value_1 = 12)
+WHERE (value_1 = 1
+       OR value_1 = 2)
 GROUP BY user_id
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_basics.out]: This query has a meaning of having at least 2 out of 3, so could you also keep third OR filter?

File: src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out [hunk 8]
```diff
@@ -120,35 +120,35 @@ FROM (
...
   ) t1 LEFT JOIN (
       SELECT DISTINCT user_id, 
         'Has done event'::TEXT AS hasdone_event
       FROM  events_table AS e
       
       WHERE 
-      (e.user_id = 20 OR e.user_id = 17)
-      AND e.event_type IN (106, 107, 108)
+      (e.user_id = 2 OR e.user_id = 3)
+      AND e.event_type IN (5)
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out]: can we keep multiple values inside 'IN'

File: src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out [hunk 9]
```diff
@@ -120,35 +120,35 @@ FROM (
...
   ) t2 ON (t1.user_id = t2.user_id)
-  WHERE (t1.user_id = 20 OR t1.user_id = 17)
+  WHERE (t1.user_id = 2)
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out]: can we keep 'OR'

File: src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out [hunk 10]
```diff
@@ -326,16 +326,16 @@ SELECT count(*), count(DISTINCT user_id), avg(user_id) FROM agg_results_second;
...
 INSERT INTO agg_results_second(user_id, value_2_agg)
 SELECT user_id, value_2 FROM users_table WHERE
-  value_2 >= 5
-  AND user_id = 96
-  AND  EXISTS (SELECT user_id FROM events_table WHERE event_type > 100 AND event_type <= 300 AND value_3 > 100 AND user_id=users_table.user_id)
-  AND  NOT EXISTS (SELECT user_id FROM events_table WHERE event_type > 300 AND event_type <= 350  AND value_3 > 100 AND user_id=users_table.user_id);
+  value_2 >= 2
+  AND user_id = 1
+  AND  EXISTS (SELECT user_id FROM events_table WHERE event_type > 1 AND event_type <= 3 AND value_3 > 1 AND user_id=users_table.user_id)
+  AND  NOT EXISTS (SELECT user_id FROM events_table WHERE event_type = 1 AND value_2 = 1 AND value_3 = 5 AND user_id=users_table.user_id);
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out]: can we keep operators intact?

File: src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out [hunk 11]
```diff
@@ -346,16 +346,16 @@ SELECT count(*), count(DISTINCT user_id), avg(user_id) FROM agg_results_second;
...
 INSERT INTO agg_results_second(user_id, value_2_agg)
 SELECT user_id, value_2 FROM users_table WHERE
-  value_2 >= 5
-  AND (user_id = 96 OR user_id = 8)
-  AND  EXISTS (SELECT user_id FROM events_table WHERE event_type > 100 AND event_type <= 300 AND value_3 > 100 AND user_id=users_table.user_id AND (user_id = 96 OR user_id = 8))
-  AND  NOT EXISTS (SELECT user_id FROM events_table WHERE event_type > 300 AND event_type <= 350  AND value_3 > 100 AND user_id=users_table.user_id AND (user_id = 96 OR user_id = 8));
+  value_2 >= 2
+  AND (user_id = 1 OR user_id = 2)
+  AND  EXISTS (SELECT user_id FROM events_table WHERE event_type > 1 AND event_type <= 3 AND value_3 > 1 AND user_id=users_table.user_id AND (user_id = 1))
+  AND  NOT EXISTS (SELECT user_id FROM events_table WHERE event_type = 1 AND value_2 = 1 AND value_3 = 5 AND user_id=users_table.user_id AND (user_id = 1));
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out]: can we keep OR?

File: src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out [hunk 12]
```diff
@@ -346,16 +346,16 @@ SELECT count(*), count(DISTINCT user_id), avg(user_id) FROM agg_results_second;
 TRUNCATE agg_results_second;
 INSERT INTO agg_results_second(user_id, value_2_agg)
 SELECT user_id, value_2 FROM users_table WHERE
-  value_2 >= 5
-  AND (user_id = 96 OR user_id = 8)
-  AND  EXISTS (SELECT user_id FROM events_table WHERE event_type > 100 AND event_type <= 300 AND value_3 > 100 AND user_id=users_table.user_id AND (user_id = 96 OR user_id = 8))
-  AND  NOT EXISTS (SELECT user_id FROM events_table WHERE event_type > 300 AND event_type <= 350  AND value_3 > 100 AND user_id=users_table.user_id AND (user_id = 96 OR user_id = 8));
+  value_2 >= 2
+  AND (user_id = 1 OR user_id = 2)
+  AND  EXISTS (SELECT user_id FROM events_table WHERE event_type > 1 AND event_type <= 3 AND value_3 > 1 AND user_id=users_table.user_id AND (user_id = 1))
```
Comment (metdos) [inline src/test/regress/expected/multi_behavioral_analytics_single_shard_queries.out]: can we keep OR?

File: src/test/regress/expected/multi_subquery_behavioral_analytics.out [hunk 13]
```diff
@@ -919,7 +883,7 @@ FROM (
     min(time) AS view_homepage_time
   FROM events_table
      WHERE user_id = 1 and
-     event_type IN (10, 20, 30, 40, 50, 60, 70, 80, 90)
+     event_type IN (1)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_behavioral_analytics.out]: can we multiple values inside `IN`?

File: src/test/regress/expected/multi_subquery_behavioral_analytics.out [hunk 14]
```diff
@@ -929,7 +893,7 @@ FROM (
   FROM events_table
   WHERE
     user_id = e1.user_id AND user_id = 1 and
-       event_type IN (11, 21, 31, 41, 51, 61, 71, 81, 91)
+       event_type IN (2)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_behavioral_analytics.out]: can we multiple values inside `IN`?

File: src/test/regress/expected/multi_subquery_behavioral_analytics.out [hunk 15]
```diff
@@ -940,7 +904,7 @@ FROM (
   FROM  events_table
   WHERE
     user_id = e2.user_id AND user_id = 1 and
-    event_type IN (12, 22, 32, 42, 52, 62, 72, 82, 92)
+    event_type IN (3)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_behavioral_analytics.out]: can we multiple values inside `IN`?

File: src/test/regress/expected/multi_subquery_behavioral_analytics.out [hunk 16]
```diff
@@ -951,7 +915,7 @@ FROM (
   FROM  events_table
   WHERE
     user_id = e3.user_id AND user_id = 1 and
-    event_type IN (13, 23, 33, 43, 53, 63, 73, 83, 93)
+    event_type IN (4)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_behavioral_analytics.out]: can we multiple values inside `IN`?

File: src/test/regress/expected/multi_subquery_behavioral_analytics.out [hunk 17]
```diff
@@ -990,7 +954,7 @@ FROM (
     min(time) AS view_homepage_time
   FROM events_table
      WHERE
-     event_type IN (10, 20, 30, 40, 50, 60, 70, 80, 90)
+     event_type IN (1)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_behavioral_analytics.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_behavioral_analytics.out [hunk 18]
```diff
@@ -1069,7 +1024,7 @@ FROM (
...
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_behavioral_analytics.out]: can we multiple values inside `IN`?

File: src/test/regress/expected/multi_subquery_complex_queries.out [hunk 19]
```diff
@@ -556,7 +556,7 @@ FROM
                     FROM 
                       events_table as  "events"
                     WHERE 
-                      event_type IN (10, 11, 12, 13, 14, 15) ) events_subquery_1) 
+                      event_type IN (1) ) events_subquery_1)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_complex_queries.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_complex_queries.out [hunk 20]
```diff
@@ -1007,15 +1002,15 @@ INNER JOIN
...
+          value_1 > 0 and value_1 < 4) AS t
         LEFT OUTER JOIN
         (
           SELECT 
             DISTINCT "events"."user_id" as user_id
           FROM 
             events_table as "events"
           WHERE 
-            event_type IN (35, 36, 37, 38)
+            event_type IN (6)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_complex_queries.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_complex_reference_clause.out [hunk 21]
```diff
@@ -516,19 +504,9 @@ FROM
...
-      14 | Tue Jan 21 05:46:51.286381 2014
-      14 | Tue Jan 21 05:46:51.286381 2014
-      14 | Tue Jan 21 05:46:51.286381 2014
-      14 | Tue Jan 21 05:46:51.286381 2014
-      14 | Tue Jan 21 05:46:51.286381 2014
-      14 | Tue Jan 21 05:46:51.286381 2014
-(10 rows)
+ user_id | lastseen 
+---------+----------
+(0 rows)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_complex_reference_clause.out]: Can we have a query which results more than  0 rows?

File: src/test/regress/expected/multi_subquery_in_where_clause.out [hunk 22]
```diff
@@ -227,7 +239,7 @@ SELECT user_id, value_2 FROM users_table WHERE
 		  FROM events_table
 		  WHERE
 		    user_id = e1.user_id AND
-		       event_type IN (11, 21, 31, 41, 51, 61, 71, 81, 91)
+		       event_type IN (2)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_in_where_clause.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_in_where_clause.out [hunk 23]
```diff
@@ -417,7 +432,7 @@ SELECT user_id, value_2 FROM users_table WHERE
...
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_in_where_clause.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_in_where_reference_clause.out [hunk 24]
```diff
@@ -187,7 +185,7 @@ SELECT user_id, value_2 FROM users_table WHERE
 		    min(time) AS view_homepage_time
 		  FROM events_reference_table
 		     WHERE
-		     event_type IN (10, 20, 30, 40, 50, 60, 70, 80, 90)
+		     event_type IN (1)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_in_where_reference_clause.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_in_where_reference_clause.out [hunk 25]
```diff
@@ -297,7 +295,7 @@ SELECT user_id, value_2 FROM users_table WHERE
         min(time) AS view_homepage_time
       FROM events_table
          WHERE
-         event_type IN (10, 20, 30, 40, 50, 60, 70, 80, 90)
+         event_type IN (1)
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_in_where_reference_clause.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_union.out [hunk 26]
```diff
@@ -658,70 +656,70 @@ FROM
...
+       1 |  53
 (5 rows)
 
 -- a long set operation list
 SELECT 
   user_id, value_3
 FROM 
 (
-  (SELECT value_3, user_id FROM events_table where event_type IN (1, 2, 3, 4, 5))
+  (SELECT value_3, user_id FROM events_table where event_type IN (1))
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_union.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_union.out [hunk 27]
```diff
@@ -658,70 +656,70 @@ FROM
...
+       6 |       2
 (5 rows)
 
 -- no partition key on the top
 SELECT 
   max(value_3)
 FROM 
 (
-  (SELECT value_3, user_id FROM events_table where event_type IN (1, 2, 3, 4, 5))
+  (SELECT value_3, user_id FROM events_table where event_type IN (1))
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_union.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_union.out [hunk 28]
```diff
@@ -877,17 +875,17 @@ SELECT
   user_id, value_3
 FROM 
 (
-  (SELECT value_3, user_id FROM events_table where event_type IN (1, 2, 3, 4, 5))
+  (SELECT value_3, user_id FROM events_table where event_type IN (1))
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_union.out]: can we multiple values inside `IN`? similarly below

File: src/test/regress/expected/multi_subquery_window_functions.out [hunk 29]
```diff
@@ -506,45 +464,45 @@ LIMIT
...
+       2 |    3
 (10 rows)
 
 -- Window functions with UNION/UNION ALL works
 SELECT
   max(avg)
 FROM
 (
-  (SELECT avg(value_3) over (partition by user_id), user_id FROM events_table where event_type IN (1, 2, 3, 4, 5))
+  (SELECT avg(value_3) over (partition by user_id), user_id FROM events_table where event_type IN (1))
```
Comment (metdos) [inline src/test/regress/expected/multi_subquery_window_functions.out]: can we multiple values inside `IN`? similarly below

**Comment by metdos** [comment]
Looks good, squash and 🚢

**Pull request merged**